modules/http-helper/dummy_server_test.go

```diff
@@ -22,6 +23,25 @@ func TestRunDummyServer(t *testing.T) {
 	HttpGetWithValidation(t, url, 200, text)
 }
 
+func TestContinuouslyCheck(t *testing.T) {
+	t.Parallel()
+
+	uniqueID := random.UniqueId()
+	text := fmt.Sprintf("dummy-server-%s", uniqueID)
+	stopChecking := make(chan bool, 1)
+
+	listener, port := RunDummyServer(t, text)
+
+	url := fmt.Sprintf("http://localhost:%d", port)
+	wg := ContinuouslyCheckUrl(t, url, stopChecking, 1*time.Second)
+	defer func() {
+		stopChecking <- true
+		wg.Wait()
+		shutDownServer(t, listener)
+	}()
```

Comment:
In theory, this test could pass if `ContinuouslyCheckUrl` was a noop... Worth adding a negative test that takes the server down part of the way through and checks that a failure happened? This might be tricky to do given that `ContinuouslyCheckUrl` calls `t.Errorf` explicitly... Unless you pass it a mock `t`...

Reply:
Ok what do you think of? 4b54663 Not perfect, but I think is slightly better.

I know we can do the error channel instead like you suggested above, but I like the fact that the test will fail without any additional logic if there is one request that failed in the current model.

Reply:
LGTM!